File: js/src/eval_exptan.js
// tan(z) = (i - i exp(2 i z)) / (exp(2 i z) + 1)

function
eval_exptan(p1)
{
	push(cadr(p1));
	evalf();
	exptan();
}

function
exptan()
{
	var p1;

	push_integer(2);
	push(imaginaryunit);
	multiply_factors(3);
	expfunc();

	p1 = pop();

	push(imaginaryunit);
	push(imaginaryunit);
	push(p1);
	multiply();
	subtract();

	push(p1);
	push_integer(1);
	add();

	divide();
}

File: src/eval_exptan.c
void
eval_exptan(struct atom *p1)
{
	scan("sqrt(-1) / (exp(2 sqrt(-1) z) + 1) - sqrt(-1) exp(2 sqrt(-1) z) / (exp(2 sqrt(-1) z) + 1)");
	push_symbol(Z_LOWER);
	push(cadr(p1));
	subst();
	evalf();
}
